alternatives.c

```diff
@@ -1290,6 +1290,41 @@ static int listServices(const char *altDir, const char *stateDir, int flags) {
     return 0;
 }
 
+int admindirExists(const char *path) {
+    struct stat sb;
+    stat(path, &sb);
+
+    if (stat(path, &sb) || !S_ISDIR(sb.st_mode) || access(path, F_OK)) {
+        if (errno == ENOENT) {
+            return 0;
+        }
+        fprintf(stderr, _("admindir %s invalid\n"), path);
+        exit(2);
+    }
+    
+    if (S_ISDIR(sb.st_mode)) {
+        return 1;
+    }
+
+    return 0;
+}
+
+int canUseAlternativeAdminDir() {
+    if (admindirExists("/var/lib/alternatives")) {
```

Comment:
If there is only one line in the block we tend to not write the {} parenthesis around it.